gfx/drivers_context/x_ctx.c

```diff
@@ -916,10 +904,20 @@ static bool gfx_ctx_x_set_video_mode(void *data,
 
                         break;
                      }
-                     else if (versions[i][0] == g_major && versions[i][1] == g_minor)
+                     else 
                      {
-                        /* The requested version was tried and is not supported, go ahead and fail since everything else will be lower than that. */
-                        break;
+                        if (versions[i][0] == g_major && versions[i][1] == g_minor)
+                        {
+                           /* last version reached, time to fail */
+                           char buf[2048];
+                           XGetErrorText(last_dpy, last_error_code, buf, sizeof buf);
```

Comment:
Sorry I should have been more clear... my suggestion was to put `XGetErrorText()` in the error handler function itself, that way, *any* X11 error is printed.

Reply:
I don't really get this...
The old `x_gl_version_error_handler` was only used for this part of the code and only effective for `glx_create_context_attribs` (aka glXCreateContextAttribsARB).
In this version, the error handling still only applies to `glx_create_context_attribs`, so what errors are there to catch?